### PR TITLE
Reconcile environment for development and docs building

### LINKED ISSUE #716: Reconcile environment for development and docs building
### Description

# Current issue

The current [environment.yml](https://github.com/pymc-devs/pytensor/blob/main/environment.yml) is not fully supported for building docs locally. The [documentation](https://pytensor.readthedocs.io/en/latest/dev_start_guide.html) for new contributor also does not mention this file yet.

# Proposals

- [x] Reconcile the file above with the file used for building doc in the workflow. Adding git+https://github.com/pymc-devs/pymc-sphinx-theme in the file is enough, I have tested locally.
- [x] Update the Developer Start Guide to use environment.yml instead.

### PR DESCRIPTION
**Pull request by HangenYuu**
## Description

Updated the [environment.yml](https://github.com/pymc-devs/pytensor/blob/main/environment.yml) for developing and docs building locally.

## Related Issue

- [ ] Closes #716 #696 
- [ ] Related to #687 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by HangenYuu** [comment]
Maybe not merge yet. Let me update the documentation first.

**Review by twiecki — APPROVED** [review]

**Comment by ricardoV94** [comment]
> Maybe not merge yet. Let me update the documentation first. I opened the PR first though.

You can click `Convert to a draft` button when a PR is not ready. That way there's no risk of premature merging. Anytime it's ready you click `Ready for review`

**Comment by HangenYuu** [comment]
Okay I have also updated the documentation. The change should be good for merging.

**Comment by OriolAbril** [comment]
Is anyone using the docgen.py script for something other than to generate the html build of the docs? I was just going over it and it basically calls sphinx with different defaults (such as default to building both html and pdf via latex instead of only building html).

If it isn't really being used (doc building and PR previews via readthedocs do not use it for example), it is probably best to remove and update the contributing guide a bit more to use the sphinx command directly instead of running docgen script

File: environment.yml [hunk 1]
```diff
@@ -37,6 +37,9 @@ dependencies:
   - pygments
   - pydot
   - ipython
+  - pip
```
Comment (OriolAbril) [inline environment.yml:40]: It can be installed from conda forge too, no need to include pip and install from github

**Review by OriolAbril — CHANGES_REQUESTED** [review]

**Comment by HangenYuu** [comment]
> Is anyone using the docgen.py script for something other than to generate the html build of the docs? I was just going over it and it basically calls sphinx with different defaults (such as default to building both html and pdf via latex instead of only building html).
> 
> If it isn't really being used (doc building and PR previews via readthedocs do not use it for example), it is probably best to remove and update the contributing guide a bit more to use the sphinx command directly instead of running docgen script

Agreed. Let me deal with this.

File: environment.yml (hunk 1 above)
Comment (OriolAbril) [reply environment.yml:40]: The one in the doc folder should also be updated then. Link: https://anaconda.org/conda-forge/pymc-sphinx-theme

**Comment by HangenYuu** [comment]
@OriolAbril 
- [x] Updated the dependency to install from `conda-forge` instead of `pip`.
- [x] Removed the script & updated the developer guide.

**Review by OriolAbril — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks a ton @HangenYuu

**Pull request merged**